### PR TITLE
Add missing license field on the OCaml compiler packages

### PR DESCRIPTION
**Pull request by kit-ty-kate**
cc @Octachron 
Partly solves https://github.com/ocaml/opam-repository/issues/18343

### CODE REVIEW

File: packages/ocaml-base-compiler/ocaml-base-compiler.4.03.0/opam [hunk 1]
```diff
@@ -1,6 +1,7 @@
 opam-version: "2.0"
 synopsis: "Official 4.03.0 release"
 maintainer: "platform@lists.ocaml.org"
+license: "QPL-1.0 AND LGPL-2.1-or-later WITH OCaml-LGPL-linking-exception"
```
Comment (Octachron) [inline packages/ocaml-base-compiler/ocaml-base-compiler.4.03.0/opam:4]: 4.03 was the first release to be distributed under the LGPL with linkink exceptions and not QPL.

**Comment by Octachron** [comment]
Normalizing the compiler opam files is a good idea.

File: packages/ocaml-base-compiler/ocaml-base-compiler.4.03.0/opam (hunk 1 above)
Comment (kit-ty-kate) [reply packages/ocaml-base-compiler/ocaml-base-compiler.4.03.0/opam:4]: are you sure? According to https://github.com/ocaml/ocaml/commit/7fc2d214c4ee8853615fc3e5649f45e11d85378d it was only done in 4.04.0. The 4.03 branch has it technically but it was only added after the release of 4.03.0

Comment (Octachron) [reply packages/ocaml-base-compiler/ocaml-base-compiler.4.03.0/opam:4]: Hm, and according to ocaml/ocaml@0ca4d6a74 the plan was to update the license for 4.03 . Keeping the license is probably more precise indeed. Sorry for the noise.

Comment (xavierleroy) [reply packages/ocaml-base-compiler/ocaml-base-compiler.4.03.0/opam:4]: I agree with @kit-ty-kate : 4.03.0 was released before the switch to LGPL.

**Review by xavierleroy — APPROVED** [review]
Thanks @kit-ty-kate for taking care of this!  I checked the main packages and it looks good to me.  Didn't check all the variant packages, but I trust you.

Side note: the line
```
authors: "Xavier Leroy and many contributors"
```
that appears in some of the `opam` files doesn't give enough justice to the "many contributors", but that's unrelated to this PR.

**Pull request merged**